src/main/java/com/monumental/services/ModelService.java

```diff
@@ -282,33 +301,41 @@ public void delete(List<Integer> ids) throws HibernateException {
     }
 
     /**
-     * Generic method to build a "FROM ... WHERE ..." query
-     * Assumes the FROM clause is the name of the T generic class
-     * @param whereParameterName - Name of the column to use in the WHERE clause
-     * @param whereParameter - Used to set the parameter in the WHERE clause
-     * @return List<T> - the List of T returned by the query
+     * Perform a get query with the specified criteria
+     * @param criteria - List of Criterion objects to apply to the query
+     * @param initializeLazyLoadedCollections - If true, loads all of the collections associated with T that are
+     *                                        normally lazy loaded
+     * @return List<T> - List of T returned by the query
      */
     @SuppressWarnings("unchecked")
-    public List<T> getFromWhere(String whereParameterName, Object whereParameter) {
+    public List<T> getWithCriteria(List<Criterion> criteria, boolean initializeLazyLoadedCollections) {
         Session session = this.sessionFactoryService.getFactory().openSession();
         Transaction transaction = null;
         List<T> records;
-        String tableName = this.getModelClass().getName();
 
         try {
             transaction = session.beginTransaction();
-            records = session.createQuery("FROM " + tableName + " WHERE " + whereParameterName + " = :whereParameter")
-                    .setParameter("whereParameter", whereParameter)
-                    .list();
+            Criteria c = session.createCriteria(this.getModelClass());
+
+            for (Criterion criterion : criteria) {
+                c.add(criterion);
+            }
```

Comment:
beauty

Reply:
I had to Google a couple of times which word was the plural and which was the singular.